### PR TITLE
DDF-3270 KMZ Input Transformer (Again)

### PR DESCRIPTION
**Pull request by gordocanchola**
This is a copy of Kevin's PR https://github.com/codice/ddf/pull/2296
This copy was made so that I can make suggested changes and handle branch conflict

#### What does this PR do?
Adds support for KMZ files using the KmlInputTransformer

#### Who is reviewing it? 
@bdeining @clockard 
(please choose AT LEAST two reviewers that need to approve the PR before it can get merged)
#### Select relevant component teams: 
https://github.com/orgs/codice/teams
#### Choose 2 committers to review/merge the PR. 
(please choose ONLY two committers from below, delete the rest)
@jlcsmith 
@pklinef 
#### What are the relevant tickets?
[](https://codice.atlassian.net/browse/)
#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests
#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by gordocanchola** [comment]
build now

**Comment by gordocanchola** [comment]
build now

**Comment by gordocanchola** [comment]
build now

**Comment by gordocanchola** [comment]
Manual testing successful.

File: catalog/spatial/kml/spatial-kml-transformer/src/test/java/org/codice/ddf/spatial/kml/transformer/TestKmzInputTransformer.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import org.junit.Test;
+
+public class TestKmzInputTransformer {
+
+  @Test
+  public void testKmzWithKmlInside() throws Exception {
+    KmzInputTransformer kmzInputTransformer =
+        new KmzInputTransformer(mock(KmlInputTransformer.class));
+    InputStream stream = TestKmzInputTransformer.class.getResourceAsStream("/kml_inside.kmz");
+    kmzInputTransformer.transform(stream);
```
Comment (jlcsmith) [inline catalog/spatial/kml/spatial-kml-transformer/src/test/java/org/codice/ddf/spatial/kml/transformer/TestKmzInputTransformer.java]: ❗️ @gordocanchola  - understand that we don't want to duplicate KMLInputTransformer tests here, but it is probably at least worth sanity checking the response

Comment (jlcsmith) [reply catalog/spatial/kml/spatial-kml-transformer/src/test/java/org/codice/ddf/spatial/kml/transformer/TestKmzInputTransformer.java]: Might want to add a test for a single KML file case as well as multiple.  Then verify that the resulting metacard matches the first KML file in the KMZ.

Comment (gordocanchola) [reply catalog/spatial/kml/spatial-kml-transformer/src/test/java/org/codice/ddf/spatial/kml/transformer/TestKmzInputTransformer.java]: Added tests

**Comment by gordocanchola** [comment]
build now

**Comment by rzwiefel** [comment]
build now

File: catalog/spatial/kml/spatial-kml-transformer/src/test/java/org/codice/ddf/spatial/kml/transformer/TestKmzInputTransformer.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+    String location = metacard.getLocation();
+    assertThat(location, not(isEmptyOrNullString()));
+    assertThat(location, containsString(EXPECTED_LAT));
+    assertThat(location, containsString(EXPECTED_LONG));
+    assertThat(location, containsString(EXPECTED_TYPE));
+  }
+
+  @Test
+  public void testKmzWithMultipleKmlAndOtherStuffInside() throws Exception {
```
Comment (coyotesqrl) [inline catalog/spatial/kml/spatial-kml-transformer/src/test/java/org/codice/ddf/spatial/kml/transformer/TestKmzInputTransformer.java]: ✏️ Would probably be good to add a test with a mocked `KmlInputTransformer` in order to ensure it is called once and only once even with a kmz file with multiple entries and that the one call is for the first entry in the zip.

**Pull request merged**